File: sdks/go/balance.go
package pushfy

import (
	"context"
	"encoding/json"
	"strconv"
)

// BalanceResource reads the account balance.
type BalanceResource struct{ c *Client }

// Balance is the parsed SMS balance. Raw is the server string (e.g. "1.500")
// and Amount is the integer value (1500).
type Balance struct {
	Raw    string
	Amount int64
}

type balanceResponse struct {
	Saldo json.Number `json:"saldo"`
}

// Get returns the SMS balance, parsing the formatted string ("1.500" -> 1500).
func (r *BalanceResource) Get(ctx context.Context) (*Balance, error) {
	var res balanceResponse
	if err := r.c.classic(ctx, "GET", "/balance", classicOpts{}, &res); err != nil {
		return nil, err
	}
	raw := res.Saldo.String()
	if raw == "" {
		return &Balance{Raw: "", Amount: 0}, nil
	}
	return &Balance{Raw: raw, Amount: parseBalance(raw)}, nil
}

// parseBalance strips every non-digit and parses the remainder ("1.500" -> 1500).
func parseBalance(raw string) int64 {
	digits := onlyDigits(raw)
	if digits == "" {
		return 0
	}
	n, err := strconv.ParseInt(digits, 10, 64)
	if err != nil {
		return 0
	}
	return n
}

File: sdks/php/src/Resource/Balance.php
<?php

declare(strict_types=1);

namespace Pushfy\Resource;

use Pushfy\Pushfy;

/** SMS balance. */
final class Balance
{
    /** @var Pushfy */
    private $client;

    public function __construct(Pushfy $client)
    {
        $this->client = $client;
    }

    /**
     * SMS balance.
     *
     * The API returns a locale-formatted string like {"saldo":"1.500"}. This
     * strips the grouping separators to an integer credit count.
     *
     * @return array{raw: string|null, balance: int|null}
     */
    public function get(): array
    {
        $res = $this->client->classicRequest('GET', '/balance');
        $raw = (is_array($res) && isset($res['saldo']) && $res['saldo'] !== null)
            ? (string) $res['saldo']
            : null;

        return [
            'raw' => $raw,
            'balance' => $raw !== null ? (int) preg_replace('/\D/', '', $raw) : null,
        ];
    }
}
